File: common/src/main/java/org/valkyrienskies/clockwork/util/blocktype/LiquidFuelType.java
package org.valkyrienskies.clockwork.util.blocktype;

import com.simibubi.create.foundation.utility.Lang;
import net.minecraft.util.StringRepresentable;
import net.minecraft.world.level.material.Fluid;
import org.valkyrienskies.clockwork.data.ClockWorkTags;

public enum LiquidFuelType implements StringRepresentable {
    NONE,
    STALE,
    PLAIN,
    SWEET,
    GOURMET,
    EXTRA
    ;


    public static LiquidFuelType byIndex(int index) {
        return values()[index];
    }

    public static LiquidFuelType fromFluid(Fluid fuel) {
        if (ClockWorkTags.AllFluidTags.STALE.matches(fuel)) {
            return LiquidFuelType.STALE;
        } else if (ClockWorkTags.AllFluidTags.PLAIN.matches(fuel)) {
            return LiquidFuelType.PLAIN;
        } else if (ClockWorkTags.AllFluidTags.SWEET.matches(fuel)) {
            return LiquidFuelType.SWEET;
        } else if (ClockWorkTags.AllFluidTags.GOURMET.matches(fuel)) {
            return LiquidFuelType.GOURMET;
        } else if (ClockWorkTags.AllFluidTags.EXTRA.matches(fuel)) {
            return LiquidFuelType.EXTRA;
        } else {
            return LiquidFuelType.NONE;
        }
    }

    public boolean isAtLeast(LiquidFuelType fuelType) {
        return this.ordinal() >= fuelType.ordinal();
    }

    @Override
    public String getSerializedName() {
        return Lang.asId(name());
    }
}

File: common/src/main/kotlin/org/valkyrienskies/clockwork/util/blocktype/LiquidFuelType.kt
package org.valkyrienskies.clockwork.util.blocktype

import com.simibubi.create.foundation.utility.Lang
import net.minecraft.util.StringRepresentable
import net.minecraft.world.level.material.Fluid
import org.valkyrienskies.clockwork.data.ClockWorkTags

enum class LiquidFuelType : StringRepresentable {
    NONE,
    STALE,
    PLAIN,
    SWEET,
    GOURMET,
    EXTRA;

    fun isAtLeast(fuelType: LiquidFuelType): Boolean {
        return ordinal >= fuelType.ordinal
    }

    override fun getSerializedName(): String {
        return Lang.asId(name)
    }

    companion object {
        fun byIndex(index: Int): LiquidFuelType {
            return values()[index]
        }

        fun fromFluid(fuel: Fluid?): LiquidFuelType {
            return if (ClockworkTags.AllFluidTags.STALE.matches(fuel)) {
                STALE
            } else if (ClockworkTags.AllFluidTags.PLAIN.matches(fuel)) {
                PLAIN
            } else if (ClockworkTags.AllFluidTags.SWEET.matches(fuel)) {
                SWEET
            } else if (ClockworkTags.AllFluidTags.GOURMET.matches(fuel)) {
                GOURMET
            } else if (ClockworkTags.AllFluidTags.EXTRA.matches(fuel)) {
                EXTRA
            } else {
                NONE
            }
        }
    }
}